jctools-core/src/main/java/org/jctools/queues/MpmcUnboundedXaddArrayQueue.java

```diff
@@ -134,6 +139,18 @@ public E poll()
                         break;
                     }
                 }
+                else if (ciChunkOffset != 0 && ccChunkIndex < ciChunkIndex)
```

Comment:
hmmm, I'm not sure I get the need for this. We're improving the performance of observing empty queue during chunk transition, which seems like an edge case, at the cost of making the code even bulkier...

Reply:
It was a case already handled in the original code to save awaiting a full rotation to complete (after `soConsumerChunk`) before ack q emptiness. 

Is an edge case, but IMO deserves to be treated correctly given that we're not using parity bits on `consumerIndex` to mark rotation states and it would save from additional pauses/spikes while (not) consuming an element. If a queue is empty I wouldn't expect to "wait" to know that, given that `isEmpty` complete in O(1).

Reply:
that's only on rotation, so an edge case. It's fine to spin on the edge case IMO

Reply:
I see that `poll' was already prone to some spinning due to the time taken by producers to reach the target chunk and that saving the check won't break the relation with `isEmpty`, but...I'm still not feeling Ok to not "fix" something a potential (rare) issue...
Probably I'm biased on this due to the pain to find a solution to this edge case :(
Hence I will trust your judgment on this.
I'm sure that if Java would support packing this:
```java
if (cIndex >= pIndex && // test against cached pIndex
    cIndex == (pIndex = lvProducerIndex())) // update pIndex if we must
{
    // strict empty check, this ensures [Queue.poll() == null iff isEmpty()]
    return null;
}
```
without creating garbage, this check should have been preserved